physics/src/modules/AtmosphereBoundaryModule/ERA5Atmosphere.cpp

```diff
@@ -46,6 +46,8 @@ ConfigurationHelp::HelpMap& ERA5Atmosphere::getHelpRecursive(HelpMap& map, bool
 
 void ERA5Atmosphere::configure()
 {
+    Finalizer::registerUnique(Module::finalize<IFluxCalculation>);
```

Comment:
just a thought... is there a way to force finalization without relying on the user to remember they have to do it?

Reply:
Almost certainly! But what does this use case look like? Which classes would be automatically finalized? I think this is a big enough piece of work on its own that is should be a separate issue where we could discuss the scope and implementation.

Reply:
My use case was more around the fact the someone may forget to register the module finalization. By default we could finalize everything at the end if it hasn't been manually triggered beforehand. Maybe I am missing something though. Happy to discuss in the next meeting

Reply:
So let's say I forget to do this step. How bad is it?

Reply:
In the CPU version? It's what `develop` is doing right now without this code, so not problem at all.
In the GPU version, my understanding is that not cleaning up data buffers before the GPU process finishes is a big headache. That is what a lot of the changes in this PR are aiming to address.